staking/parse.go

```diff
@@ -136,18 +136,8 @@ func parsePublicKey(oid asn1.ObjectIdentifier, publicKey asn1.BitString) (crypto
 			return nil, ErrInvalidRSAPublicExponent
 		}
 
-		if pub.N.Sign() <= 0 {
```

Comment:
> Are lines 129:137 being called in the call chain of clients of ValidateRSAPublicKeyIsWellFormed?

I am not sure I follow, I just exported a few lines to a method because I want to call them from another method - `ValidateRSACertificate` in `tls_config.go`.

> Why is it necessary for the code moved to ValidateRSAPublicKeyIsWellFormed not performed by that same call chain?

I also don't quite understand what you're asking, can you be more clear? Which call chain are you referring to, exactly?

I exported the method to avoid code duplication and get better code coverage via tests.

Reply:
> Is the validation on the lines mentioned in my previous comment (L129-137) performed by the golang tls connection code? If so, presumably that same stdlib codepath is not performing the extra validation represented by ValidateRSAPublicKeyIsWellFormed - why not?


Right, so all this change is doing is essentially adding some checks to the TLS handshake such as not supporting huge keys (8192 bits) or with an exponent that is not the default exponent of Golang's TLS stack.
You can see in my test that is indeed does not do these checks, as the error we get are returned by code that I added, and not by code that was already there in the Go TLS layer. 

As for why not - well I don't know, but I do know that previous versions of avalanchego also didn't support such keys. All we're doing here is making the checks more efficient - by moving them to the TLS handshake itself and not to after it had already happened.

Reply:
> As for why not - well I don't know, but I do know that previous versions of avalanchego also didn't support such keys.

These restrictions were imposed to ensure that "all reasonable RSA implementations would agree on the validity of the signatures with reasonable performance."